sdk/servicebus/azure-messaging-servicebus-track1-perf/src/main/java/com/azure/messaging/servicebus/perf/ReceiveAndLockMessageTest.java

```diff
@@ -0,0 +1,110 @@
+package com.azure.messaging.servicebus.perf;
+
+import com.azure.messaging.servicebus.perf.core.ServiceBusStressOptions;
+import com.azure.messaging.servicebus.perf.core.ServiceTest;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Collection;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+
+public class ReceiveAndLockMessageTest extends ServiceTest<ServiceBusStressOptions> {
+
+    public ReceiveAndLockMessageTest(ServiceBusStressOptions options) throws InterruptedException, ExecutionException, ServiceBusException {
+        super(options, ReceiveMode.PEEKLOCK);
+    }
+
+    public Mono<Void> globalSetupAsync() {
+        // Since test does warm up and test many times, we are sending many messages, so we will have them available.
+        int totalMessageMultiplier = 50;
+
+        String messageId = UUID.randomUUID().toString();
+        Message message = new Message(CONTENTS);
+        message.setMessageId(messageId);
+
+        return Flux.range(0, options.getMessagesToSend() * totalMessageMultiplier)
+            .flatMap(count -> {
```

Comment:
Why are we sending one at a time instead of using sending an iterable? You can use an IntStream to create all the messages.